src/components/Form/Controls/SettingsToggle/SettingsToggle.tsx

```diff
@@ -0,0 +1,79 @@
+/*
+Copyright 2025 New Vector Ltd.
+
+SPDX-License-Identifier: AGPL-3.0-only OR LicenseRef-Element-Commercial
+Please see LICENSE files in the repository root for full details.
+*/
+
+import {
+  ComponentProps,
+  ComponentRef,
+  forwardRef,
+  PropsWithChildren,
+} from "react";
+import { ToggleInput } from "../Toggle";
+import { Label } from "../../Label";
+import { HelpMessage } from "../../Message";
+import { InlineField } from "../../InlineField";
+import React from "react";
+import { Control } from "@radix-ui/react-form";
+
+type SettingsToggleProps = {
+  /**
+   * The CSS class name for the containing field.
+   */
+  className?: string;
+  /**
+   * The field name.
+   */
+  name: string;
+  label: string;
+  /**
+   * Optional help text to display below the setting.
+   */
+  helpMessage?: string;
+  /**
+   * Optional help text to display below the setting.
+   */
+  disabledMessage?: string;
+} & Omit<ComponentProps<typeof ToggleInput>, "type">;
+
+/**
+ * A toggle component.
+ */
+export const SettingsToggleInput = forwardRef<
+  HTMLInputElement,
+  PropsWithChildren<SettingsToggleProps>
+>(function Toggle(
+  { className, label, helpMessage, disabledMessage, name, ...props },
+  ref,
+) {
+  const content = (
+    <InlineField
+      className={className}
+      name={name}
+      control={<ToggleInput ref={ref} {...props} />}
+    >
+      <Label>{label}</Label>
+      {helpMessage && <HelpMessage>{helpMessage}</HelpMessage>}
+      {disabledMessage && props.disabled && (
+        <HelpMessage>{disabledMessage}</HelpMessage>
+      )}
+    </InlineField>
+  );
+  return content;
```

Comment:
Sorry to complain after it being merged, but I'm failing to see the value of that kind of component 😅 

I'm really in favour of using composition in React rather than defining everything as props. This is how you end up with components with 100s of props.

All the base components here already existed to support this exact use case, so why the need to define a *very specific component* here?